.github/workflows/reusable_run_test_and_examples_by_linking_type.yml

```diff
@@ -67,12 +67,12 @@ jobs:
       - name: Build Tests & Examples
         run: |
           cmake -G Ninja -B build_dir -DBUILD_TESTING=ON -DBUILD_EXAMPLES=ON \
-          ${{ fromJSON('["", "-DBUILD_SHARED_LIBS=ON -DCMAKE_POSITION_INDEPENDENT_CODE=ON"]')[inputs.build-shared-libs] }} \
-          ${{ fromJSON('["", "-DCMAKE_CXX_FLAGS=--coverage"]')[inputs.with-coverage] }}  
+          ${{ fromJSON('["", "-DBUILD_SHARED_LIBS=ON -DCMAKE_POSITION_INDEPENDENT_CODE=ON"]')[inputs.build-shared-libs] }}
           cmake --build build_dir
         env:
           CC: ${{ steps.install_cc.outputs.cc }}
           CXX: ${{ steps.install_cc.outputs.cxx }}
+          CXXFLAGS: -fsanitize=address -fsanitize=undefined ${{ fromJSON('["", "--coverage"]')[inputs.with-coverage] }}
```

Comment:
clang suggests to use clang++ for linking and to keep function frame pointers

Reply:
Added `-fno-omit-frame-pointer -Wno-unused-command-line-argument -fuse-ld=lld`

`-Wno-unused-command-line-argument` is just that clang does complain on every cpp file getting _compiled_ that it doesn't use the linker, which is expected anyways at compile stage.